lib/customer.rb

```diff
@@ -0,0 +1,38 @@
+require "csv"
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    data = CSV.read("./data/customers.csv")
+    customer_info = []
+
+    data.each do |row|
+      id = row[0].to_i
+      email = row[1]
+      address = {}
+      address[:street] = row[2]
+      address[:city] = row[3]
+      address[:state] = row[4]
+      address[:zip] = row[5]
+      
+      customer = Customer.new(id, email, address)
+      customer_info << customer
+    end
+
+    return customer_info
+  end
+
+  def self.find(id)
+    customer_info = self.all
```

Comment:
Instead of saying `self.all` here, you could say just `all` ... but `self.all` is a perfectly reasonable choice, and may even make your purpose extra clear.